src/engine/blocks.js

```diff
@@ -91,21 +91,21 @@ Blocks.prototype.getOpcode = function (id) {
 
 /**
  * Get all fields and their values for a block.
- * @param {?string} id ID of block to query.
- * @return {!object} All fields and their values.
+ * @param {?object} block The block to query.
+ * @return {?object} All fields and their values.
  */
-Blocks.prototype.getFields = function (id) {
-    var block = this._blocks[id];
+Blocks.prototype.getFields = function (block) {
+    // var block = this._blocks[id];
     return (typeof block === 'undefined') ? null : block.fields;
 };
 
 /**
  * Get all non-branch inputs for a block.
- * @param {?string} id ID of block to query.
+ * @param {?object} block the block to query.
  * @return {!object} All non-branch inputs and their associated blocks.
  */
-Blocks.prototype.getInputs = function (id) {
-    var block = this._blocks[id];
+Blocks.prototype.getInputs = function (block) {
```

Comment:
I bet eventually it may also make sense to have this/the branch inputs separated on block-representation-construction-time, rather than determining it inline